src/js/utils/option-utils.jsx

```diff
@@ -27,10 +27,11 @@ export const debounceUsersFetch = (waitTime, minSearchLength) =>
     }
   }, waitTime);
 
-export const debounceLocationsFetch = (waitTime, minSearchLength) =>
+export const debounceLocationsFetch = (waitTime, minSearchLength, type) =>
   _.debounce((searchTerm, callback) => {
     if (searchTerm && searchTerm.length >= minSearchLength) {
-      apiClient.get(`/openboxes/api/locations?direction=${queryString.parse(window.location.search).direction}&name=${searchTerm}`)
+      const { direction } = queryString.parse(window.location.search);
+      apiClient.get(`/openboxes/api/locations?name=${searchTerm}${direction ? `&direction=${direction}` : ''}${type ? `&type=${type}` : ''}`)
```

Comment:
This part is trickier for me since my brain struggles with functional languages. Not sure how to add an array as a parameter, but something like this
```
var activityCodesParams = activityCodes.map(function(activityCode){ return "activityCode=" + activityCode; }).join("&")
```